presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -0,0 +1,703 @@
+            resultBuilder.put(entry.getKey(), Optional.ofNullable(partition));
+        }
+        return resultBuilder.build();
+    }
+
+    private List<Partition> batchGetPartition(String databaseName, String tableName, List<String> partitionNames)
+    {
+        try {
+            List<PartitionValueList> partitionValueLists = partitionNames.stream()
+                    .map(p -> new PartitionValueList().withValues(toPartitionValues(p))).collect(toList());
+
+            List<List<PartitionValueList>> batchedPartitionValueLists = Lists.partition(partitionValueLists, BATCH_GET_PARTITION_MAX_PAGE_SIZE);
+            List<Future<BatchGetPartitionResult>> batchGetPartitionFutures = new ArrayList<>();
+            List<Partition> result = new ArrayList<>();
+
+            for (List<PartitionValueList> partitions : batchedPartitionValueLists) {
+                batchGetPartitionFutures.add(glueClient.batchGetPartitionAsync(new BatchGetPartitionRequest()
+                        .withDatabaseName(databaseName)
+                        .withTableName(tableName)
+                        .withPartitionsToGet(partitions)));
+            }
+
+            for (Future<BatchGetPartitionResult> future : batchGetPartitionFutures) {
+                BatchGetPartitionResult batchResult = future.get();
+                batchResult.getPartitions().stream().forEach(p -> result.add(GlueToPrestoConverter.convertPartition(p)));
+            }
+
+            return result;
+        }
+        catch (AmazonServiceException | InterruptedException | ExecutionException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public void addPartitions(String databaseName, String tableName, List<Partition> partitions)
+    {
+        try {
+            List<List<Partition>> batchedPartitions = Lists.partition(partitions, BATCH_CREATE_PARTITION_MAX_PAGE_SIZE);
+            List<Future<BatchCreatePartitionResult>> futures = new ArrayList<>();
+
+            for (List<Partition> partitionBatch : batchedPartitions) {
+                List<PartitionInput> partitionInputs = partitionBatch.stream().map(GlueInputConverter::convertPartition).collect(toList());
+                futures.add(glueClient.batchCreatePartitionAsync(new BatchCreatePartitionRequest()
+                        .withDatabaseName(databaseName)
+                        .withTableName(tableName)
+                        .withPartitionInputList(partitionInputs)));
+            }
+
+            for (Future<BatchCreatePartitionResult> future : futures) {
+                BatchCreatePartitionResult result = future.get();
+                propagatePartitionErrorToPrestoException(databaseName, tableName, result.getErrors());
+            }
+        }
+        catch (AmazonServiceException | InterruptedException | ExecutionException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
```

Comment:
Restore the thread interrupt status on interrupted:
```java
if (e instanceof InterruptedException) {
    Thread.currentThread().interrupt();
}

Reply:
Will add